requirements/dev.txt

```diff
@@ -79,7 +79,7 @@ django-waffle==3.0.0
     # via
     #   edx-django-utils
     #   edx-drf-extensions
-djangorestframework==3.13.1
+djangorestframework==3.14.0
```

Comment:
This change [removes the deprecated `NullBooleanField`](https://github.com/encode/django-rest-framework/pull/8599/commits/d6b87b0b0ea847d00156936908124051908874d3), which is easy to work around in this code.

But it introduces an [issue with the installed version of drf-yasg](https://github.com/axnsan12/drf-yasg/issues/810) which is resolved by [drf-yasg==1.21.4](https://github.com/axnsan12/drf-yasg/releases/tag/1.21.4), whose version is managed by [edx/api-doc-tools](https://github.com/openedx/api-doc-tools/pull/155/files#diff-869ddaf7a3f1194d76b3152b5c430bfa2b96e24b5aab47da0670458131518683R71).

So we need to wait for their next round of automated dependency updates to update ^ and submit a new dependency PR here too.